src/Components/ColorPalette/ColorPalette.jsx

```diff
@@ -1,42 +1,82 @@
 import "../../css/main.css";
-import "./colorPalette.css";
+import "./colorpalette.css";
 import { useState } from "react";
 
-export const ColorPalette = ({setInputCardDetails, inputCardDetails}) => {
-    const colors = [
-        "#FAF9F6",
-        "#f28983",
-        "#fbbc04",
-        "#FFF475",
-        "#CCFF90",
-        "#A7FFEB",
-        "#CBF0F8",
-        "#AECBFA",
-        "#D7AEFB",
-        "#FDCFE8",
-        "#E6C9A8",
-        "#E8EAED"
-      ];
+export const ColorPalette = ({
+  setInputCardDetails,
+  inputCardDetails,
+  edit,
+  editCardDetails,
+  setEditCardDetails,
+}) => {
+  const colors = [
+    "#FAF9F6",
+    "#f28983",
+    "#fbbc04",
+    "#FFF475",
+    "#CCFF90",
+    "#A7FFEB",
+    "#CBF0F8",
+    "#AECBFA",
+    "#D7AEFB",
+    "#FDCFE8",
+    "#E6C9A8",
+    "#E8EAED",
+  ];
 
-    const [paletteDisplay, setPaletteDisplay] = useState(false);
-    return(
-        <div className="button-palette"
-                onMouseOver={()=>{setPaletteDisplay(!paletteDisplay)}}
-                onMouseOut={()=>{setPaletteDisplay(!paletteDisplay)}}
-                style={{color: paletteDisplay ? "var(--primary-color)" : "black"}}>
-            <span className="material-icons-outlined">palette</span>
-        
-            <div className={`${paletteDisplay ? "container-palette flex flex-gap-0-5 flex-wrap flex-justify-space-around" : "display-none"}`} >
-                {
-                    colors.map((color, index) => {
-                        return(
-                            <div key={index} className="palette" style={{backgroundColor: color, border: inputCardDetails.selectedBackgroundColor===color ? "2px solid black" : "none"}}
-                                onClick={() => {setInputCardDetails({...inputCardDetails, selectedBackgroundColor: color})}}></div>
-                        )
-                    } )
-                }       
-            </div>
+  const [paletteDisplay, setPaletteDisplay] = useState(false);
+  const { isEdit } = edit;
 
-        </div>
-    )
-}
\ No newline at end of file
+  return (
+    <div
+      className="button-palette"
+      onMouseOver={() => {
+        setPaletteDisplay(!paletteDisplay);
+      }}
+      onMouseOut={() => {
+        setPaletteDisplay(!paletteDisplay);
+      }}
+      style={{ color: paletteDisplay ? "var(--primary-color)" : "black" }}
+    >
+      <span className="material-icons-outlined">palette</span>
+
+      <div
+        className={`${
+          paletteDisplay
+            ? "container-palette flex flex-gap-0-5 flex-wrap flex-justify-space-around"
+            : "display-none"
+        }`}
+      >
+        {colors.map((color, index) => {
+          return (
+            <div
+              key={index}
+              className="palette"
+              style={{
+                backgroundColor: color,
+                border: !isEdit
+                  ? inputCardDetails.selectedBackgroundColor === color
+                    ? "2px solid black"
```

Comment:
suggestion: Don't write styles between jsx it's not best practice

Reply:
It is very difficult to use different classes for different colors here , so I thought for this scenario in line styling would be the best. For border I agree I can use a class and remove it here but I think I cannot remove the inline style for background color. If you have any idea how I could implement this without inline styles please let me know.